Source/Public/Get-VSTeamUserEntitlement.ps1

```diff
@@ -9,22 +9,57 @@ function Get-VSTeamUserEntitlement {
       [int] $Skip = 0,
 
       [Parameter(ParameterSetName = 'List')]
+      [Parameter(ParameterSetName = 'PagedFilter')]
+      [Parameter(ParameterSetName = 'PagedParams')]
       [ValidateSet('Projects', 'Extensions', 'Grouprules')]
       [string[]] $Select,
 
       [Parameter(ParameterSetName = 'ByID')]
       [Alias('UserId')]
-      [string[]] $Id
+      [string[]] $Id,
+
+      [Parameter(ParameterSetName = 'PagedFilter')]
+      [Parameter(ParameterSetName = 'PagedParams')]
+      [int] $MaxPages = 0,
+
+      [Parameter(ParameterSetName = 'PagedFilter')]
+      [string] $Filter,
+
+      [Parameter(ParameterSetName = 'PagedParams')]
+      [ValidateSet('Account-Stakeholder', 'Account-Express', 'Account-Advanced', IgnoreCase = $false)]
+      [Alias('License')]
+      [string] $LicenseId,                                                            #license name is case sensitive
+
+      [Parameter(ParameterSetName = 'PagedParams')]
+      [ValidateSet('guest', 'member', IgnoreCase = $false)]                            #userType is case sensitive: values in lowercase
+      [string] $UserType,
+
+      [Parameter(ParameterSetName = 'PagedParams')]
+      [Alias('UserName')]
+      [Alias('Mail')]
+      [string] $Name
    )
 
+
    process {
-      # This will throw if this account does not support MemberEntitlementManagement
-      _supportsMemberEntitlementManagement
+      # This will throw if this account does not support MemberEntitlementManagement 
+      # or supported version is not correct with the type of API call
+      $paramCounter = _countParameters -BoundParameters $PSBoundParameters
+      $paramset = 'PagedParams', 'PagedFilter'
+      if ($paramCounter -eq 0) {
+         _supportsMemberEntitlementManagement

```

Comment:
Yes, my first idea was having into account until 5.1 version or above. 

But there is a scenario, when invoking the cmdlet without parameters, that should work on both versions. That is the reason for the 0 parameters check. Then I realized that I used common argumetns like -Verbose, the 0 parameters check fails. And that is the reason I created the _countParameters.

Also, as this function is very specific I put it in the local file instead of moving it the the common functions.

Reply:
Ok. I leave it like it is here. Currently I don't see a more elegant way.